Add simple test to load People screen

mokagio opened the pull request:
This relates to the issue reported in https://github.com/wordpress-mobile/WordPress-iOS/issues/20112.

I was hoping to use this to test #20114, but the test _doesn't actually catch the crash_. When the People screen loads from the UI tests, it does not crash but gets stuck in some kind of loading loop where the animation continually restarts.

I decided to not investigate this further because the issue with the loading loop is likely due to outdated or incomplete API request JSON stubs and I don't have the time at this point to chase that down

I decide to keep the test anyway because it's a valuable starting point to ensure that the people screen is available and interactive.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – This PR adds UI tests :)

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Add accessibility identifier to "People" row in My Site

mokagio committed:
Add simple test to load People screen

This relates to the issue reported in
https://github.com/wordpress-mobile/WordPress-iOS/issues/20112.

The test _doesn't actually catch the crash_.
When the People screen loads from the UI tests, it gets stuck in some
kind of loading loop where the animation continually restarts.

I decided to not investigate this further because the issue with the
loading loop is likely due to outdated or incomplete API request JSON
stubs and I don't have the time at this point to chase that down

I decide to keep the test anyway because it's a valuable starting point
to ensure that the people screen is available and interactive.

mokagio commented on WordPress/WordPressUITests/Tests/MainNavigationTests.swift:
```diff
@@ -17,26 +17,41 @@ class MainNavigationTests: XCTestCase {
+    // We run into an issue where the People screen would crash short after loading.
+    // See https://github.com/wordpress-mobile/WordPress-iOS/issues/20112.
+    //
+    // It would be wise to add similar tests for each item in the menu (then remove this comment).
+    func testLoadsPeopleScreen() throws {
         XCTAssert(MySiteScreen.isLoaded(), "MySitesScreen screen isn't loaded.")
 
-        _ = try mySiteScreen
-            .tabBar.goToReaderScreen()
+        mySiteScreen.goToMenu()
+        mySiteScreen.goToPeople()
 
-        XCTAssert(ReaderScreen.isLoaded(), "Reader screen isn't loaded.")
+        // We don't have a ScreenObject for the people screen,
+        // so we just check that the previous screen is not loaded
+        XCTAssertFalse(MySiteScreen.isLoaded(), "MySitesScreen screen isn't loaded.")
+    }
```
The diff got mangled, but this is the only real change: A new `testLoadsPeopleScreen()` method.

I did not touch the other one.

crazytonyli commented on WordPress/WordPressUITests/Tests/MainNavigationTests.swift:
```diff
@@ -17,26 +17,41 @@ class MainNavigationTests: XCTestCase {
-        XCTAssert(ReaderScreen.isLoaded(), "Reader screen isn't loaded.")
+        // We don't have a ScreenObject for the people screen,
+        // so we just check that the previous screen is not loaded
+        XCTAssertFalse(MySiteScreen.isLoaded(), "MySitesScreen screen isn't loaded.")
```
I think this would fail on iPad, since the "My Site" page is still visible? Maybe we can assert the tabs("Users", "Folowers", etc) on the People screen exist instead? Or adding an accessibility identifier to the view controller, so that we can find it in the view hierarchy in the UI test and assert it's loaded?

crazytonyli replied:
[The iPad test](https://buildkite.com/automattic/wordpress-ios/builds/12326#018639d6-9e67-4921-a128-e668aabebe04/597-4248) indeed failed.

mokagio replied:
Oh, interesting... 🤔 I'll have to look into that.

jostnes commented on WordPress/WordPressUITests/Tests/MainNavigationTests.swift:
```diff
@@ -17,26 +17,41 @@ class MainNavigationTests: XCTestCase {
 
-        _ = try mySiteScreen
-            .tabBar.goToReaderScreen()
+        mySiteScreen.goToMenu()
```
Looks like it's failing on iPad because of this line. Since on iPad, the menu items are already listed on screen (including the "People" item), this isn't required anymore. What do you think about adding a condition for iPhone?

```suggestion
        if XCUIDevice.isPhone {
            mySiteScreen.goToMenu()
        }
```

jostnes commented on WordPress/WordPressUITests/Tests/MainNavigationTests.swift:
```diff
@@ -17,26 +17,41 @@ class MainNavigationTests: XCTestCase {
-        _ = try mySiteScreen
-            .tabBar.goToReaderScreen()
+        mySiteScreen.goToMenu()
+        mySiteScreen.goToPeople()
```
And I think you're right in that the reason clicking on "People" item results in a loading loop is because of missing stubs:

![image](https://user-images.githubusercontent.com/17252150/218395388-5801fbeb-267b-488a-be2c-5810ff94bbf6.png)

mokagio committed:
Account for iPhone/iPad differences in `goToMenu()`

mokagio committed:
Add People screen for UI tests

mokagio committed:
Address a SwiftLint violation

Autocorrect would have generated this diff:

```
@@ -7,9 +7,7 @@ public class PeopleScreen: ScreenObject {
         try super.init(
             expectedElementGetters: [
                 // See the PeopleViewController.Filter rawValues
-                { $0.buttons["users"] },
-                { $0.buttons["followers"] },
-                { $0.buttons["email"] },
+                { $0.buttons["users"] }, { $0.buttons["followers"] }, { $0.buttons["email"] },
             ],
             app: app,
             waitTimeout: 7

```

I didn't like it and I also didn't want to add a skip annotation, this
refactor adds more lines of code but it's tool agnostic.
